From c9c76f243e4b04dfbb4879594e2900babd024b8e Mon Sep 17 00:00:00 2001
From: Srujana Challa <schalla@marvell.com>
Date: Mon, 28 Sep 2020 19:24:11 +0530
Subject: [PATCH 0912/1921] cn10k-cpt: forward ready message to AF

Fixes rclk and sclk settings in MBOX_MSG_READY
for CPT VF.

Signed-off-by: Srujana Challa <schalla@marvell.com>
Change-Id: I1058eaa79cd322643b912418bf1380bc6b018986
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/36866
Reviewed-by: Anoob Joseph <anoobj@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 .../crypto/marvell/cn10k/cn10k_cptpf_mbox.c   | 22 -------------------
 1 file changed, 22 deletions(-)

diff --git a/drivers/crypto/marvell/cn10k/cn10k_cptpf_mbox.c b/drivers/crypto/marvell/cn10k/cn10k_cptpf_mbox.c
index c76e32e9a522..0b76c77aef74 100644
--- a/drivers/crypto/marvell/cn10k/cn10k_cptpf_mbox.c
+++ b/drivers/crypto/marvell/cn10k/cn10k_cptpf_mbox.c
@@ -243,24 +243,6 @@ static int check_attach_rsrcs_req(struct cn10k_cptpf_dev *cptpf,
 	return forward_to_af(cptpf, vf, req, size);
 }
 
-static int reply_ready_msg(struct cn10k_cptpf_dev *cptpf,
-			   struct cn10k_cptvf_info *vf,
-			   struct mbox_msghdr *req)
-{
-	struct ready_msg_rsp *rsp;
-
-	rsp = (struct ready_msg_rsp *)
-	       otx2_mbox_alloc_msg(&cptpf->vfpf_mbox, vf->vf_id, sizeof(*rsp));
-	if (!rsp)
-		return -ENOMEM;
-
-	rsp->hdr.id = MBOX_MSG_READY;
-	rsp->hdr.sig = OTX2_MBOX_RSP_SIG;
-	rsp->hdr.pcifunc = req->pcifunc;
-
-	return 0;
-}
-
 static int reply_eng_grp_num_msg(struct cn10k_cptpf_dev *cptpf,
 				 struct cn10k_cptvf_info *vf,
 				 struct mbox_msghdr *req)
@@ -440,10 +422,6 @@ static int cptpf_handle_vf_req(struct cn10k_cptpf_dev *cptpf,
 		return otx2_reply_invalid_msg(&cptpf->vfpf_mbox, vf->vf_id,
 					      req->pcifunc, req->id);
 	switch (req->id) {
-	case MBOX_MSG_READY:
-		err = reply_ready_msg(cptpf, vf, req);
-		break;
-
 	case MBOX_MSG_ATTACH_RESOURCES:
 		err = check_attach_rsrcs_req(cptpf, vf, req, size);
 		break;
-- 
2.31.1

